src/test/java/uk/gov/ons/ctp/response/casesvc/service/CaseServiceTest.java

```diff
@@ -713,6 +713,60 @@ public void testEventRespondentEnrolled() throws Exception {
         .postAction(any(String.class), any(UUID.class), any(String.class));
   }
 
+  /**
+   * Tests that with multiple survey ids then the limit per survey matches maxCasesPerSurvey and
+   * that the later cases in the list are rejected This suffices because SpringBoot implements the
+   * findByDateTime desc , so we only need validate the later ones are removed
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testLimitCasesPerSurveyRemovesCasesFromEndOfList() throws Exception {
+
+    int maxCasesPerSurvey = 2;
+
+    // Arrange: Create a List<Case> of 6 entries corresponding to two groups
+
+    UUID groupUUIDOne = UUID.randomUUID();
+    UUID groupUUIDTwo = UUID.randomUUID();
+
+    Case case_one = makeCase(groupUUIDOne);
+    Case case_two = makeCase(groupUUIDOne);
+    Case case_three = makeCase(groupUUIDOne);
+    Case case_four = makeCase(groupUUIDTwo);
+    Case case_five = makeCase(groupUUIDTwo);
+    Case case_six = makeCase(groupUUIDTwo);
+
+    List<Case> caseList = new ArrayList<>();
+
+    caseList.add(case_one);
+    caseList.add(case_two);
+    caseList.add(case_three);
+    caseList.add(case_four);
+    caseList.add(case_five);
+    caseList.add(case_six);
+
+    CaseGroup caseGroupOne = makeCaseGroupWithSurveyId(UUID.randomUUID());
+    CaseGroup caseGroupTwo = makeCaseGroupWithSurveyId(UUID.randomUUID());
+
+    when(caseGroupRepo.findById(groupUUIDOne)).thenReturn(caseGroupOne);
+    when(caseGroupRepo.findById(groupUUIDTwo)).thenReturn(caseGroupTwo);
+
+    // Act: limit those to 2 cases per survey id
+
+    List<Case> results = caseService.limitCasesPerSurvey(caseList, maxCasesPerSurvey);
```

Comment:
As far as I can tell this is the only method that uses the public nature of `limitCasesPerSurvey` - the usual way of testing this as a private method would be to mock out `caseRepo.findByPartyIdOrderByCreatedDateTimeDesc` like you have just above this and then calling the public method `findCasesByPartyIdLimitedPerSurvey`, thereby testing the whole lot rather than just the filtering method.

Reply:
Its one of those odd ones in that I wanted a unit test (possibly need a higher level one as well)  But couldn't as it was private . You could use this function to modify any list of cases per survey so it could be a useful utility. Protected is possibly a better match.